WooCommerce/Classes/ViewRelated/Products/AI/ProductSharingMessageGenerationViewModel.swift

```diff
@@ -84,6 +89,15 @@ final class ProductSharingMessageGenerationViewModel: ObservableObject {
         }
         analytics.track(event: .ProductSharingAI.shareButtonTapped(withMessage: messageContent.isNotEmpty))
     }
+
+    /// Handles when a feedback is sent.
+    func handleFeedback(_ vote: FeedbackView.Vote) {
+        analytics.track(event: .AIFeedback.feedbackSent(source: .productSharingMessage,
+                                                        isUseful: vote == .up))
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) { [weak self] in
```

Comment:
❓ Why do we have this delay? Could you also kindly add a comment?

Reply:
It's basically for the better UX effect - to show the state change of the tapped button before making the banner disappear. Added comments in d5031fd2339b3d3e8d8dc4ca3aaf20c7f5b629bf.